addons/account/models/chart_template.py

```diff
@@ -636,6 +636,21 @@ def _post_load_data(self, template_code, company, template_data):
             company.account_purchase_tax_id = self.env['account.tax'].search([
                 *self.env['account.tax']._check_company_domain(company),
                 ('type_tax_use', 'in', ('purchase', 'all'))], limit=1).id
+        # Set default taxes on products (only on products having already a tax set in another company, as some flows require no tax at all (e.g TIPS in PoS))
+        all_companies = self.env['res.company'].sudo().search([]).ids
+        # need to get all companies ids and browse the product in sudo + allowed_company_ids to check for the taxes_id and supplier_taxes_id fields
+        # that would just look empty all the time for the current user/company
+        sudoed_products = self.env['product.product'].sudo().with_context(allowed_company_ids=all_companies).search(self.env['product.product']._check_company_domain(company))
```

Comment:
Doesn't `sudo` give access to all companies since it's not checking the record rules? :thinking: 
```suggestion
        sudoed_products = self.env['product.product'].sudo().search(self.env['product.product']._check_company_domain(company))
```

Then you don't need to get `all_companies` at all

Reply:
Would you be able to explain why?
It is very counter intuitive to me; I tested and it seems to be working just fine.

Reply:
Might be invalidation issues :thinking: 
The product has only one default tax, for every single company
```python
In [1]: self.env(user=3).user.name
Out[1]: 'Default User Template'

In [2]: self.env(user=3).user.company_ids
Out[2]: res.company(1,)

In [3]: self.env(user=3)['product.product'].browse(1).taxes_id
Out[3]: account.tax(1,)

In [4]: self.env(user=3)['product.product'].sudo().browse(1).taxes_id
Out[4]: account.tax(1,)

In [5]: self.env.invalidate_all()

In [6]: self.env(user=3)['product.product'].sudo().browse(1).taxes_id
Out[6]: account.tax(1, 3, 109, 141)

In [7]: self.env(user=3)['product.product'].browse(1).taxes_id
Out[7]: account.tax(1, 3, 109, 141)
```

Reply:
you were right, I removed the allowed_companies :) :+1: